### PR TITLE
feat: expand develop section, closes #1955

### LINKED ISSUE #1955: Document .taurignore in develop section
> > This definitely needs to be documented for people using Rust workspace.
> 
> I agree but where would you expect to find this in the docs? Genuine question as we're currently rewriting the docs and i have no idea where to put this.

I would imagine a sub-section or a separate page under `develop` such as here:

https://beta.tauri.app/guides/develop/

Which is about the DX experience, including things like setting up IDEs for Tauri dev., where best to place shared env variables and keys, how to set up auto-reloads (where `.taurignore` fits), etc.

_Originally posted by @boan-anbo in https://github.com/tauri-apps/tauri/issues/4617#issuecomment-1992197832_

### PR DESCRIPTION
**Pull request by lucasfernog**
Expand develop overview, including mobile guides.

### CODE REVIEW

**Comment by mhedgpeth** [comment]
I'm new to tauri and read through your changes and found them helpful. I learned a lot. Much appreciated.

**Review by dreyfus92 — APPROVED** [review]
Thank you @lucasfernog, LGTM ✅

File: src/content/docs/develop/index.mdx [hunk 1]
```diff
@@ -46,14 +65,138 @@ for a more detailed explanation.
...
+  npm="npm run tauri [android|ios] dev --open"
+  yarn="yarn tauri [android|ios] dev --open"
+  pnpm="pnpm tauri [android|ios] dev --open"
+  cargo="cargo tauri [android|ios] dev --open"
+/>
+
+:::caution
+To use Xcode or Android Studio the Tauri CLI process **must** be running and **cannot** be killed.
+It is recommended to use the `tauri [android|ios] dev --open` command and keep the process alive until you close the IDE.
+:::
```
Comment (ghost) [inline src/content/docs/develop/index.mdx]: I think it's also worth noting, in the case of XCode at least, that the `Build Rust Code` phase is ran within a sanitized environment that might not include the tauri binary location, resulting in a fail. (May happen depending on how the toolchain was installed, see [#6573](https://github.com/tauri-apps/tauri/issues/6573))

Easiest fix I found for me using cargo was to symlink the cargo binary to `/usr/local/bin/cargo`

Comment (lucasfernog) [reply src/content/docs/develop/index.mdx]: will do, thanks!

**Pull request merged**